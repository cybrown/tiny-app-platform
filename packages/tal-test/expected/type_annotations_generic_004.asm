main()
  entry:
    FunctionRef       name: box_0
    DeclareLocal      name: box, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Literal           42
    Literal           1
    MakeArray
    Literal           0
    MakeRecord
    Local             box
    Call
    Pop               inBlock: false
    Literal           42
    Literal           1
    MakeArray
    Literal           0
    MakeRecord
    Local             box
    Call
box_0(item)
  entry:
    Literal           "value"
    Local             item
    Literal           1
    MakeRecord
    MakeArrayForBlock count: 1
