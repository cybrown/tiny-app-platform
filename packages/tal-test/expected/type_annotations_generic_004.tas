fun box(item: T): {value: T} { {value: item} }

box(42)
box(42)
