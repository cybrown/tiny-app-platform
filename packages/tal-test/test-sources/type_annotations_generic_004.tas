fun box<T>(item: T): {value: T} {
    {value: item}
}

box<T: number, A: string>(42)
box<T: number>(42)
