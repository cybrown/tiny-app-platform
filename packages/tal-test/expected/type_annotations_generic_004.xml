<program>
  <declarelocal>
    <function>
      <block>
        <record>
          <recordentry>
            <local />
          </recordentry>
        </record>
      </block>
    </function>
  </declarelocal>
  <call>
    <local />
    <literal />
  </call>
  <call>
    <local />
    <literal />
  </call>
</program>
